src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationTransport.java

```diff
@@ -413,6 +416,25 @@ private JSONArray createTestsResultsJSONArray(Collection<TestResults> testResult
         return testResultsArray;
     }
 
+    /**
+     * Creates an JSONArray containing task names and their state (Success, Failed, Error)
+     *
+     * @param taskResults Collection of all defined tasks with details
+     * @return JSONArray containing the name and state
+     * @throws JSONException
+     */
+    private JSONArray createTasksJSONArray(Collection<TaskResult> taskResults) throws JSONException {
+        logToBuildLog("Creating tasks JSON array");
+        JSONArray tasksArray = new JSONArray();
+        for (TaskResult taskResult : taskResults) {
+            JSONObject taskJSON = new JSONObject();
+            taskJSON.put("name", taskResult.getTaskIdentifier().getUserDescription());
```

Comment:
can we add any other details that might help us later on? For example what kind of task this is? Any additional options that are specified in the task?

Reply:
Added the pluginKey, isEnabled and isFinal